compressors/external_compressors.py

```diff
@@ -0,0 +1,142 @@
+"""External compressors like gzip, etc. for testing/benchmarking purposes
+"""
+
+import os
+import tempfile
+from core.data_block import DataBlock
+from core.data_encoder_decoder import DataDecoder, DataEncoder
+from core.data_stream import TextFileDataStream, Uint8FileDataStream
+from core.encoded_stream import EncodedBlockReader, EncodedBlockWriter
+from core.prob_dist import ProbabilityDist
+from utils.bitarray_utils import BitArray, bitarray_to_uint, uint_to_bitarray
+from utils.test_utils import (
+    create_random_binary_file,
+    try_file_lossless_compression,
+    try_lossless_compression,
+)
+import zlib
+
+
+class ZlibExternalEncoder(DataEncoder):
```

Comment:
Out of curiosity what does Zlib use under the hood? Maybe add a link etc. to Zlib documentation here and in the decoder

Reply:
added comment in revision

Reply:
thanks!